Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -595,14 +607,23 @@ namespace AzToolsFramework
         m_handlerCleanupTimer->setSingleShot(true);
         m_handlerCleanupTimer->setInterval(0);
         connect(m_handlerCleanupTimer, &QTimer::timeout, this, &DocumentPropertyEditor::CleanupReleasedHandlers);
+
+        m_spawnDebugView = ed_showDPEDebugView;
     }
 
     DocumentPropertyEditor::~DocumentPropertyEditor()
     {
     }
 
-    void DocumentPropertyEditor::SetAdapter(AZ::DocumentPropertyEditor::DocumentAdapter* theAdapter)
+    void DocumentPropertyEditor::SetAdapter(AZ::DocumentPropertyEditor::DocumentAdapterPtr theAdapter)
     {
+        if (m_spawnDebugView)
+        {
+            QPointer<AzToolsFramework::DPEDebugWindow> theWindow = new AzToolsFramework::DPEDebugWindow(nullptr);
+            theWindow->SetAdapter(theAdapter);
+            theWindow->show();
+        }
```

Comment:
It might be better to add a button for popping up this window instead of always spawning it, especially once you start dealing with integrating with the Entity Inspector (where you'd get N debug views per component)